File: src/main/java/tk/hiddenname/smarthome/exception/PinSignalSupportException.java
package tk.hiddenname.smarthome.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(value = HttpStatus.CONFLICT)
public class PinSignalSupportException extends Exception{

    public PinSignalSupportException(int gpio) {
        super(String.format("Gpio '%d' doesn't support this type of signal", gpio));
    }
}

File: src/main/java/tk/hiddenname/smarthome/exception/PinSignalSupportException.kt
package tk.hiddenname.smarthome.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(value = HttpStatus.CONFLICT)
class PinSignalSupportException(gpioPin: Int) : Exception("GPIO pin '$gpioPin' doesn't support this type of signal")